src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseGitDiffReportResource.java

```diff
@@ -128,61 +131,26 @@ public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForSu
-        participationAuthCheckService.checkCanAccessParticipationElseThrow(submission2.getParticipation());
-        var report = gitDiffReportService.generateReportForSubmissions(submission1, submission2);
-        return ResponseEntity.ok(new ProgrammingExerciseGitDiffReportDTO(report));
-    }
-
-    /**
-     * GET exercises/:exerciseId/submissions/:submissionId1/diff-report-commit-details-with-template : Get the diff report for a submission of a programming exercise with the
-     * template of the exercise. To fetch the diff report for the commit details view, the current user needs to have access to the participation of the submission.
-     *
-     * @param exerciseId    the id of the exercise the submission and the template belong to
-     * @param submissionId1 the id of the submission
-     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with the diff report as body
-     * @throws GitAPIException if errors occur while accessing the git repository
-     * @throws IOException     if errors occur while accessing the file system
-     */
-    @GetMapping("programming-exercises/{exerciseId}/submissions/{submissionId1}/diff-report-commit-details-with-template")
-    @EnforceAtLeastStudent
-    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommitDetailsViewForSubmissionWithTemplate(@PathVariable long exerciseId,
-            @PathVariable long submissionId1) throws GitAPIException, IOException {
-        log.debug("REST request to get a ProgrammingExerciseGitDiffReport for the commit details view for submission {} with the template of exercise {}", submissionId1,
-                exerciseId);
-        var exercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
-        var submission = submissionRepository.findById(submissionId1).orElseThrow();
-        if (!submission.getParticipation().getExercise().getId().equals(exerciseId)) {
-            throw new ConflictException("A git diff report entry can only be retrieved if the submissions exercise id matches with the exercise id", ENTITY_NAME,
-                    "exerciseIdsMismatch");
-        }
-        participationAuthCheckService.checkCanAccessParticipationElseThrow(submission.getParticipation());
-        var report = gitDiffReportService.createReportForSubmissionWithTemplate(exercise, submission);
+    public ResponseEntity<ProgrammingExerciseGitDiffReportDTO> getGitDiffReportForCommits(@PathVariable long participationId, @PathVariable String commitHash1,
+            @PathVariable String commitHash2) throws GitAPIException, IOException {
+        log.debug("REST request to get a ProgrammingExerciseGitDiffReport for two commits for commit {} and commit {} of participation {}", commitHash1, commitHash2,
+                participationId);
+
+        ProgrammingExerciseStudentParticipation participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
```

Comment:
Missing consistency-check between the exerciseId and the exercise of the participation.